analyzer/tests/unit/test_log_parser.py

```diff
@@ -253,3 +254,119 @@ def test_compiler_implicit_include_flags_sysroot(self):
         flags = ["-I", "/usr/include", "--sysroot=/usr/mysysroot"]
         filtered = log_parser.filter_compiler_includes_extra_args(flags)
         self.assertEqual(filtered, ["--sysroot=/usr/mysysroot"])
+
+    def test_skip_everything_from_parse(self):
+        """Same skip file for pre analysis and analysis. Skip everything."""
+        cmp_cmd_json = [
+            {"directory": "/tmp/lib1",
+             "command": "g++ /tmp/lib1/a.cpp",
+             "file": "a.cpp"},
+            {"directory": "/tmp/lib1",
+             "command": "g++ /tmp/lib1/b.cpp",
+             "file": "b.cpp"},
+            {"directory": "/tmp/lib2",
+             "command": "g++ /tmp/lib2/a.cpp",
+             "file": "a.cpp"}]
+
+        skip_list = """
+        -*/lib1/*
+        -*/lib2/*
+        """
+        analysis_skip = skiplist_handler.SkipListHandler(skip_list)
+        pre_analysis_skip = skiplist_handler.SkipListHandler(skip_list)
+
+        build_actions = log_parser.\
+            parse_unique_log(cmp_cmd_json, self.__this_dir,
+                             analysis_skip_handler=analysis_skip,
+                             pre_analysis_skip_handler=pre_analysis_skip)
+
+        self.assertEqual(len(build_actions), 0)
+
+    def test_skip_all_in_pre_from_parse(self):
+        """Pre analysis skips everything but keep build action for analysis."""
+        cmp_cmd_json = [
+            {"directory": "/tmp/lib1",
+             "command": "g++ /tmp/lib1/a.cpp",
+             "file": "a.cpp"},
+            {"directory": "/tmp/lib1",
+             "command": "g++ /tmp/lib1/b.cpp",
+             "file": "b.cpp"},
+            {"directory": "/tmp/lib2",
+             "command": "g++ /tmp/lib2/a.cpp",
+             "file": "a.cpp"}]
+
+        keep = {"directory": "/tmp/lib2",
```

Comment:
It can be equal with `cmp_cmd_json[2]`:
```suggestion
        keep = cmp_cmd_json[2]
```